packages/data-context/src/data/coreDataShape.ts

```diff
@@ -138,6 +138,10 @@ export interface CoreDataShape {
   packageManager: typeof PACKAGE_MANAGERS[number]
   forceReconfigureProject: ForceReconfigureProjectDataShape | null
   cancelActiveLogin: (() => void) | null
+  versionData: {
+    latestVersion?: Promise<string | undefined>
+    npmMetadata?: Promise<Record<string, string> | undefined>
```

Comment:
nit: I really don't like these tri-states, it's a decent amount of mental load when `latestVersion` can either be `undefined`, `Promise<undefined>`, or `Promise<string>`. I try to refactor them whenever I encounter them. Can you think of a way to clean this up?

Reply:
We could probably get it to: 
```
latestVersion?: Promise<string>,
npmMetadata?: Promise<Record<string, string>>
```
let me give it a shot

Reply:
@tgriesser no luck here?